classes/Spcies_options.m

```diff
@@ -0,0 +1,699 @@
+%% Spcies_options - Spcies class for defining and working with the toolbox options
+%
+% This class contains all the options of the toolbox and methods for working with them,
+% getting default options, etc.
+% 
+% This class is part of Spcies: https://github.com/GepocUS/Spcies
+% 
+
+classdef Spcies_options < matlab.mixin.Copyable
+
+    %% PROPERTIES
+    properties
+
+        % User selection
+        formulation {ischar} % Stores the name of the formulation 
+        method {ischar} % Stores the name of the optimization method
+        submethod {ischar} % Stores the name of the optimization submethod
+        platform {ischar} % Stores the name of the target platform
+
+        % Toolbox options
+        verbose % Determines the verbose level of the toolbox (currently unused)
+
+        % General options for code generation
+        save_name {ischar} % Stores the name used as seed for file generation
+        directory {ischar} % Stores the directory used as seed for file generation
+        override % Determines if files with the same name are overwritten
+        const_are_static % Determines if const in C are also declared as static variables
+        precision {ischar} % Determines if 'double' of 'float' precision is used for real numbers
+        inf_value % Determines the number assigned to "infinity"
+        save % Determines if files are saved (currently unused, but included for backwards-compatibility for now)
+
+        % General solver options (used in all, or nearly all, solvers)
+        debug % Determines if the solver is built in debug-mode (returns additional info to user)
+        timing % Determines if the solver measures and returns computation times
+        in_engineering % If true, then the solver considers the inputs/outputs to be in engineering units
+        time_varying % If true, then the solver considers the system and MPC ingredients to be time-varying
+        force_diagonal % Tells the solver to consider ingredients as diagonal if it can
+
+    end
+
+    properties (Hidden=false, SetAccess=protected)
```

Comment:
If we want to be able to change the options inside the structure when we are computing ingredients, which is the case in compute_MPCT_EADMM_ingredients.m, we would have to change "SetAccess=protected" to "SetAccess=public".

If you want to maintain this to "SetAccess=protected" for any reason, please let me know and then I can find another way to implement the case of MPCT with EADMM when rho is scalar but without changing the internal values of the options structure.